applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_HCF_analysis.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics
+import KratosMultiphysics.StructuralMechanicsApplication as SMA
+
+# Importing the base class
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_analysis import StructuralMechanicsAnalysis
+
+class StructuralMechanicsHCFAnalysis(StructuralMechanicsAnalysis):
+    """
+    This class is used to complement the structurea_mechanics_analysis 
+    
+    when using the HCF constitutive law
+    """
+    def __init__(self, model, project_parameters):
+        super(StructuralMechanicsHCFAnalysis, self).__init__(model, project_parameters)
+
+    def RunSolutionLoop(self):
+        """This function executes the solution loop of the AnalysisStage
+        It can be overridden by derived classes
+        """
+        while self.KeepAdvancingSolutionLoop():
+            process = SMA.AdvanceInTimeStrategyHighCycleFatigueProcess(self._GetSolver().GetComputingModelPart(), self.project_parameters)
+            if self.project_parameters["fatigue"]["advancing_strategy"].GetBool():
+                process.Execute()
+                time_incr = self._GetSolver().GetComputingModelPart().ProcessInfo[SMA.TIME_INCREMENT]
+                self.time += time_incr
+                self._GetSolver().GetComputingModelPart().ProcessInfo[SMA.TIME_INCREMENT] = 0.0
+
+            self._GetSolver().GetComputingModelPart().ProcessInfo[KratosMultiphysics.TIME] = self.time
```

Comment:
can you please explain what is happening here? I am a bit lost

Reply:
> can you please explain what is happening here? I am a bit lost

I have modified a bit this. The idea is that for the HCF and advance-in-time process is required to avoid losing time in the linear domain. Thus, when the appropiate conditions are satisfied, the time variables, i.e. time and the number of cycles being applied are modified. This is what is done here.

Reply:
> > can you please explain what is happening here? I am a bit lost
> 
> I have modified a bit this. The idea is that for the HCF and advance-in-time process is required to avoid losing time in the linear domain. Thus, when the appropiate conditions are satisfied, the time variables, i.e. time and the number of cycles being applied are modified. This is what is done here.

Tell me if this was what you asked for or if you want me to go more in detail.